mavimage/chunkedbytes.py

```diff
@@ -1,42 +1,120 @@
+                                    (index*self._chunk_size+self._chunk_size)]
+            return item
+        else:
+            raise ValueError("Error: Can only accept an integer value")
 
-    def __getitem__(self, item):
-        # item: bytes
-        return item
-
-    def __setitem__(self, key, value):
-        # key, value
+    def __setitem__(self, key, byte_sequence):
+        """set an item in the byte sequence
+        Args:
+            key: integer
+            byte_sequence: bytes
+        """
+        first_part = self._bytes_item[:(key*self._chunk_size)]
+        last_part = self._bytes_item[((key*self._chunk_size)+self._chunk_size):]
+        self._bytes_item = first_part + byte_sequence + last_part
 
     def __delitem__(self, key):
-        # key
+        """Delete a whole index of the of the ChunkedBytes based on chunk size
+        Args:
+            key: integer
+        """
+        first_part = self._bytes_item[:(key * self._chunk_size)]
+        last_part = self._bytes_item[((key * self._chunk_size) + self._chunk_size):]
+        self._bytes_item = first_part + last_part
 
     def __len__(self):
-        length = 0
-        # length: int
-        return length
+        """Find length of indeces of the bytes based on chunk size
+        Returns:
+            length: int
+        """
+        return math.ceil(len(self._bytes_item)/self._chunk_size)
 
     def __add__(self, other):
-        # other: bytes or ChunkedBytes
+        """Add bytes or ChunkedBytes. Else raise a NotImplemented exception
+        Args:
+            other: bytes or ChunkedBytes
+        Returns:
+            bytes
+        """
+        if isinstance(other, bytes):
+            self._bytes_item = self._bytes_item + other
+        else:
+            try:
+                self._bytes_item = self._bytes_item + other._bytes_item
+            except AttributeError:
+                raise NotImplementedError
+        return self._bytes_item
 
     def __radd__(self, other):
-        # other: bytes or ChunkedBytes
+        """Add with bytes or ChunkedBytes on right of operator.
+        Else raise a NotImplemented exception
+        Args:
+            other: bytes or ChunkedBytes
+        Returns:
+            bytes
+            """
+        if isinstance(other, bytes):
+            self._bytes_item = other + self._bytes_item
+        else:
+            try:
+                self._bytes_item = other._bytes_item + self._bytes_item
+            except AttributeError:
+                raise NotImplementedError
+        return self._bytes_item
 
     def zeros(self, chunks, chunk_size=256):
-        # chunks: int, chunk_size: int
+        """Return zeros of length chunks*chunk_size
+        Args:
+             chunks: int
+             chunk_size: int
+        """
+        zero_array = b"0" * chunk_size
+        for i in range(0, chunks-1):
+            zero_array = zero_array + (b"0" * chunk_size)
+        return zero_array
```

Comment:
This is setting the bytes to the byte representing the character 0. which is byte '\x60' (a numeric value of 48).  This should set to the 0 value byte (null byte) which is '\x00'.